packages/flutter/lib/src/rendering/box.dart

```diff
@@ -2360,42 +2558,105 @@ abstract class RenderBox extends RenderObject {
-      _cachedBaselines?.clear();
-      _cachedIntrinsicDimensions?.clear();
-      _cachedDryLayoutSizes?.clear();
+  void _debugVerifyDryBaselines() {
+    assert(() {
+      final List<DiagnosticsNode> messages = <DiagnosticsNode>[
+        ErrorDescription(
+          'The constraints used were $constraints.',
+        ),
+        ErrorHint(
+          'If you are not writing your own RenderBox subclass, then this is not\n'
+          'your fault. Contact support: https://github.com/flutter/flutter/issues/new?template=2_bug.yml',
+        )
+      ];
+
+      // Only perform the baseline checks after performLayout. Currently this
+      // method is also run in the RenderBox.size setter and after
+      // performResize, and it's not safe to access the baseline location in
+      // those two places since it may depend on the child layout.
+      for (final TextBaseline baseline in TextBaseline.values) {
+        assert(!RenderObject.debugCheckingIntrinsics);
+        RenderObject.debugCheckingIntrinsics = true;
+        _debugDryLayoutCalculationValid = true;
+        final double? dryBaseline;
+        final double? realBaseline;
+        try {
+          dryBaseline = getDryBaseline(constraints, baseline);
+          realBaseline = getDistanceToBaseline(baseline, onlyReal: true);
+        } finally {
+          RenderObject.debugCheckingIntrinsics = false;
+        }
+        assert(!RenderObject.debugCheckingIntrinsics);
+        if (!_debugDryLayoutCalculationValid || dryBaseline == realBaseline) {
+          continue;
+        }
+        if ((dryBaseline == null) != (realBaseline == null)) {
+          final (String methodReturnedNull, String methodReturnedNonNull) = dryBaseline == null
+            ? ('computeDryBaseline', 'computeDistanceToActualBaseline')
+            : ('computeDistanceToActualBaseline', 'computeDryBaseline');
+          throw FlutterError.fromParts(<DiagnosticsNode>[
+            ErrorSummary(
+              'The $baseline location returned by ${objectRuntimeType(this, 'RenderBox')}.computeDistanceToActualBaseline '
+              'differs from the baseline location computed by computeDryBaseline.'
+            ),
+            ErrorDescription(
+              'The $methodReturnedNull method returned null while the $methodReturnedNonNull returned a non-null $baseline of ${dryBaseline ?? realBaseline}. '
+              'Did you forget to implement $methodReturnedNull for ${objectRuntimeType(this, 'RenderBox')}?'
+            ),
+            ...messages,
+          ]);
+        } else {
+          throw FlutterError.fromParts(<DiagnosticsNode>[
+            ErrorSummary(
+              'The $baseline location returned by ${objectRuntimeType(this, 'RenderBox')}.computeDistanceToActualBaseline '
+              'differs from the baseline location computed by computeDryBaseline.'
+            ),
+            DiagnosticsProperty<RenderObject>(
+              'The RenderBox was',
+              this,
+            ),
```

Comment:
Wondering, why this isn't part of the previous error message?

Reply:
I think the previous case is to catch when people forget to implement `computeDistanceToActualBaseline` or `computeDryBaseline`. So which RenderBox instance doesn't really matter, we only have to give people the runtime type of the problematic `RenderBox` subclass?